core/src/module/usModuleContext.cpp

```diff
@@ -175,8 +177,14 @@ std::string ModuleContext::GetDataFile(const std::string &filename) const
 #else
     static const char separator = '/';
 #endif
-
-  std::string baseStoragePath = ModuleSettings::GetStoragePath();
+  
+  std::string baseStoragePath;
+  std::map<std::string, std::string>::iterator prop = d->module->coreCtx->frameworkProperties.find("org.osgi.framework.storage");
```

Comment:
Use something like Framework::STORAGE_LOCATION (look up the OSGi standard for a naming suggestion) instead of the raw string literal.